### PR TITLE
replace margo pkg with updated mochi-margo pkg

### PR DESCRIPTION
**Pull request by carns**
- advance from v0.4.3 to v0.9 of margo
- change spack package name to "mochi-margo" to match preferred naming convention
- add stub mochi package as placeholder for deprecated name

### CODE REVIEW

**Comment by becker33** [comment]
I think this PR should modify all builtin packages that depend on margo as well.

**Comment by carns** [comment]
> I think this PR should modify all builtin packages that depend on margo as well.

I would be happy to (there is just one package in the spack builtin repo that depends on margo: unifyfs).  I'll defer to @CamStan and others on that team if they want that change to be included in this PR, though.

**Comment by CamStan** [comment]
I'm happy to update the unifyfs package myself for this change. Currently have a PR open (#20271) to update unifyfs and I can always update the package name to mochi-margo there once this is merged.

However, we'll probably have another change coming to the unifyfs package soon as we test to make sure we work with the latest versions of Mochi-margo and Mercury. So I could make the change then as well.

**Comment by adamjstewart** [comment]
We've never created a dummy package after a package rename in the past, so we should decide if this is how we want to do things. I opened https://github.com/spack/spack/discussions/15896 a while back but so far it hasn't gotten much attention.

**Comment by carns** [comment]
> We've never created a dummy package after a package rename in the past, so we should decide if this is how we want to do things. I opened #15896 a while back but so far it hasn't gotten much attention.

Ok.  I don't have a preference either way; we can drop the dummy package if that makes things easier (that name is not widely used; most of our user base uses an external repo with the "mochi-margo" name that we prefer).

**Comment by carns** [comment]
Just checking in on this; I know the timing wasn't great on this PR over the holiday break.

Anything I can do to help push it through?  Myself and my team have no preference on the dummy package issue, so I'm happy to revise that aspect or anything else that might make the PR more palatable.  Thanks!

**Comment by adamjstewart** [comment]
#15896 and #20767 have gained a bit of traction. For now, let's keep `margo` the way it used to be (basically a duplicate of `mochi-margo`, not a dummy package). When #20767 is merged, we can officially deprecate the old package. Does that sound good to you?

**Comment by carns** [comment]
Thanks @adamjstewart , that sounds great to me.  I'll rebase and adjust this PR accordingly shortly.  I also subscribed to that discussion and PR so that I can help with deprecating later when that functionality is available.

File: var/spack/repos/builtin/packages/margo/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/margo/package.py]: Should we keep @fbudin69500 @chuckatkins as maintainers/add them to `mochi-margo`?

File: var/spack/repos/builtin/packages/margo/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/margo/package.py]: Instead of using tags, which are moving targets in git, could we use URLs and checksums for all versions (for both packages)?

File: var/spack/repos/builtin/packages/margo/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/margo/package.py]:
```suggestion
    version('develop', branch='develop')
```

File: var/spack/repos/builtin/packages/margo/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/margo/package.py]:
```suggestion
    depends_on('pkgconfig', type=("build"))
```
Spack has a virtual package for this

Comment (carns) [reply var/spack/repos/builtin/packages/margo/package.py]: Thanks @adamjstewart , I'll make these changes later today.  I'll probably drop the @develop version here and just leave the @master one, though; there is no develop branch in the source repo.

Comment (carns) [reply var/spack/repos/builtin/packages/margo/package.py]: I just pushed the requested revisions @adamjstewart I just pushed the revisions if you want to have a look.

Comment (carns) [reply var/spack/repos/builtin/packages/margo/package.py]: whoops modulo formatting goof, now I got it :)

Comment (carns) [reply var/spack/repos/builtin/packages/margo/package.py]: Actually there is a CI problem here that I don't fully understand https://github.com/spack/spack/pull/20273/checks?check_run_id=1749955042

**Comment by adamjstewart** [comment]
@trws can you take a look at the `mypy` failure? This is the same one I see locally from time-to-time.

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/margo/package.py
Comment (haampie) [inline var/spack/repos/builtin/packages/margo/package.py]:
```suggestion
# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
```

**Review by haampie — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/margo/package.py
Comment (haampie) [reply var/spack/repos/builtin/packages/margo/package.py]: This should be fixed on the `develop` branch, let me close and reopen to trigger ci again

**Comment by carns** [comment]
Thanks all. It looks like the tests are good now.  Any other blockers?  (the copyright date change is a-ok, of course, but I don't think my review is needed on that)

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
